gstreamer/src/query.rs

```diff
@@ -260,32 +348,53 @@ macro_rules! declare_concrete_query(
             }
         }
 
+        impl<'a> Deref for $name<&'a mut QueryRef> {
+            type Target = $name<&'a QueryRef>;
+
+            fn deref(&self) -> &Self::Target {
+                unsafe {
+                    mem::transmute(self)
+                }
+            }
+        }
+
         impl<'a> $name<&'a mut QueryRef> {
             pub fn get_mut_query(&mut self) -> &mut QueryRef {
                 self.0
             }
         }
 
-        impl<'a> Deref for $name<&'a mut QueryRef> {
-            type Target = $name<&'a QueryRef>;
+        impl Deref for $name<Query> {
+            type Target = QueryRef;
 
             fn deref(&self) -> &Self::Target {
-                unsafe {
-                    mem::transmute(self)
-                }
+                &self.0
+            }
+        }
+
+        impl DerefMut for $name<Query> {
+            fn deref_mut(&mut self) -> &mut Self::Target {
+                assert!(self.0.is_writable());
+                self.0.make_mut()
```

Comment:
This can (and probably should be) `self.0.make_mut().unwrap()`. It can't possibly fail, it does not need to go through `gst_query_make_writable()`.

Reply:
get_mut I mean, not make. Sorry